networkit/cpp/distance/BidirectionalBFS.cpp

```diff
@@ -0,0 +1,108 @@
+#include <limits>
+
+#include <networkit/distance/BidirectionalBFS.hpp>
+
+namespace NetworKit {
+
+void BidirectionalBFS::run() {
+    if (G.isWeighted())
+        WARN("Treating the graph as unweighted!");
+
+    stDist = 0;
+    if (source == target) {
+        hasRun = true;
+        return;
+    }
+
+    init();
+    if (storePred) {
+        pred[source] = source;
+        pred[target] = target;
+    }
+
+    // Two queues per BFS: sQueue(next) for the exploration from the source node, tQueue(next) for
+    // the exploration from the target node.
+    std::queue<node> sQueue, sQueueNext, tQueue, tQueueNext;
```

Comment:
The BFS class allocates its objects locally, while the Dijkstra class stores them in the class. I would suggest to also store them in the class for BFS.